osu.Game/Overlays/Chat/ChannelControl/ControlItem.cs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics.Containers;
+using osu.Game.Online.Chat;
+
+namespace osu.Game.Overlays.Chat.ChannelControl
+{
+    public class ControlItem : OsuClickableContainer
+    {
+        public event Action<Channel>? OnRequestSelect;
+        public event Action<Channel>? OnRequestLeave;
+
+        public int MentionCount
+        {
+            get => mention?.MentionCount ?? 0;
+            set
+            {
+                if (mention != null)
+                    mention.MentionCount = value;
+            }
+        }
+
+        public bool HasUnread
+        {
+            get => text?.HasUnread ?? false;
+            set
+            {
+                if (text != null)
+                    text.HasUnread = value;
+            }
+        }
+
+        private Box? hoverBox;
+        private Box? selectBox;
+        private ControlItemText? text;
+        private ControlItemMention? mention;
+        private ControlItemClose? close;
+
+        [Resolved]
+        private Bindable<Channel>? selectedChannel { get; set; }
+
+        [Resolved]
+        private OverlayColourProvider? colourProvider { get; set; }
+
+        private readonly Channel channel;
+
+        public ControlItem(Channel channel)
+        {
+            this.channel = channel;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            Height = 30;
+            RelativeSizeAxes = Axes.X;
+
+            Children = new Drawable[]
+            {
+                hoverBox = new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = colourProvider!.Background3,
+                    Alpha = 0f,
+                },
+                selectBox = new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = colourProvider!.Background4,
+                    Alpha = 0f,
+                },
+                new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Padding = new MarginPadding { Left = 18, Right = 5 },
```

Comment:
Not sure this padding should be allowed to be asymmetric like this, it looks pretty bad to me. Would make it symmetric.

Reply:
Given the reference design doesn't show any channel text reaching the extent of the right padding, I took cues from Discord's design for this asymmetrical padding.
I think having it symmetrical will make the text look too cluttered, especially since the mention number and the close button push the text in from the right.
I agree it looks too close on the right side though, so I've made that side larger a bit.